model/mention_map.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"fmt"
+	"net/url"
+)
+
+type UserMentionMap map[string]string
+type ChannelMentionMap map[string]string
+
+const (
+	UserMentionsKey       = "user_mentions"
+	UserMentionsIdsKey    = "user_mentions_ids"
+	ChannelMentionsKey    = "channel_mentions"
+	ChannelMentionsIdsKey = "channel_mentions_ids"
+)
+
+func UserMentionMapFromURLValues(values url.Values) (UserMentionMap, error) {
+	return mentionsFromURLValues(values, UserMentionsKey, UserMentionsIdsKey)
+}
+
+func (m UserMentionMap) ToURLValues() url.Values {
+	return mentionsToURLValues(m, UserMentionsKey, UserMentionsIdsKey)
+}
+func ChannelMentionMapFromURLValues(values url.Values) (ChannelMentionMap, error) {
```

Comment:
nit: suggest newline for clarity
```suggestion

func ChannelMentionMapFromURLValues(values url.Values) (ChannelMentionMap, error) {
```

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/13762/commits/552c67e6d459e98b646fd149c792a2bb0eb7486d.